src/python/pants/backend/go/lint/vet/rules_integration_test.py

```diff
@@ -0,0 +1,160 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+from textwrap import dedent
+
+import pytest
+
+from pants.backend.go import target_type_rules
+from pants.backend.go.lint import fmt
+from pants.backend.go.lint.vet import skip_field
+from pants.backend.go.lint.vet.rules import GoVetFieldSet, GoVetRequest
+from pants.backend.go.lint.vet.rules import rules as go_vet_rules
+from pants.backend.go.lint.vet.subsystem import GoVetSubsystem
+from pants.backend.go.target_types import GoModTarget, GoPackageTarget
+from pants.backend.go.util_rules import (
+    assembly,
+    build_pkg,
+    first_party_pkg,
+    go_mod,
+    import_analysis,
+    link,
+    sdk,
+    third_party_pkg,
+)
+from pants.core.goals.lint import LintResult, LintResults
+from pants.core.util_rules import source_files
+from pants.engine.addresses import Address
+from pants.engine.fs import CreateDigest, Digest, FileContent
+from pants.engine.rules import SubsystemRule
+from pants.engine.target import Target
+from pants.testutil.rule_runner import QueryRule, RuleRunner
+
+
+@pytest.fixture()
+def rule_runner() -> RuleRunner:
+    rule_runner = RuleRunner(
+        target_types=[GoModTarget, GoPackageTarget],
+        rules=[
+            *fmt.rules(),
+            *skip_field.rules(),
+            *go_vet_rules(),
+            *source_files.rules(),
+            *target_type_rules.rules(),
+            *first_party_pkg.rules(),
+            *third_party_pkg.rules(),
+            *sdk.rules(),
+            *go_mod.rules(),
+            *import_analysis.rules(),
+            *link.rules(),
+            *build_pkg.rules(),
+            *assembly.rules(),
+            QueryRule(LintResults, (GoVetRequest,)),
+            SubsystemRule(GoVetSubsystem),
+        ],
+    )
+    rule_runner.set_options(
+        ["--backend-packages=pants.backend.experimental.go.lint.vet"], env_inherit={"PATH"}
```

Comment:
FYI it's not necessary to set `--backend-packages`. That's a bad pattern I introduced into the repo

Reply:
Also you already call `set_options` inside `run_go_vet`, which will override this. So I think you can remove this line.

Reply:
Good point. Removed.

Reply:
Had to add it back, got this error:

```
E           pants.engine.internals.scheduler.ExecutionError: 1 Exception encountered:
E
E           Engine traceback:
E             in select
E             in pants.engine.internals.graph.resolve_target (//:mod#./)
E             in pants.backend.go.target_type_rules.generate_targets_from_go_mod
E             in pants.backend.go.util_rules.go_mod.determine_go_mod_info (//:mod)
E             in pants.engine.process.fallible_to_exec_result_or_raise
E             in pants.engine.process.upcast_process
E             in pants.backend.go.util_rules.sdk.setup_go_sdk_process
E             in pants.backend.go.util_rules.sdk.go_sdk_invoke_setup
E             in pants.backend.go.subsystems.golang.setup_goroot
E           Traceback (most recent call last):
E             File "/private/var/folders/md/0q71p41n0rbgwhnc8npjjzg00000gn/T/process-executionUh1Gn7/src/python/pants/engine/internals/selectors.py", line 705, in native_engine_generator_send
E               res = func.send(arg)
E             File "/private/var/folders/md/0q71p41n0rbgwhnc8npjjzg00000gn/T/process-executionUh1Gn7/src/python/pants/backend/go/subsystems/golang.py", line 100, in setup_goroot
E               "Cannot find any `go` binaries using the option "
E           pants.engine.process.BinaryNotFoundError: Cannot find any `go` binaries using the option `[golang].go_search_paths`: []
E
E           To fix, please install Go (https://golang.org/doc/install) with the version 1.17 (set by `[golang].expected_version`) and ensure that it is discoverable via `[golang].go_search_paths`.
```